File: Language/C#/Book/tmp/LINQ实战/Chapter04.CSharp/Chapter04.Win/FormStrings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;

namespace LinqInAction.Chapter04.Win
{
  public partial class FormStrings : Form
  {
    public FormStrings()
    {
      InitializeComponent();
    }

    private void FormStrings_Load(object sender, EventArgs e)
    {
      String[] books = { "Funny Stories", "All your base are belong to us",
        "LINQ rules", "C# on Rails", "Bonjour mon Amour" };

      var query =
        from book in books
        where book.Length > 10
        orderby book
        select new { Book=book.ToUpper() };

      dataGridView1.DataSource = query.ToList();
    }
  }
}

File: Language/C#/Book/LINQ实战/第04章 熟悉LINQ to Objects/Chapter04.VB.Win/FormStrings.vb
Public Class FormStrings

  Private Sub FormStrings_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim books As String() = {"Funny Stories", "All your base are belong to us", _
          "LINQ rules", "C# on Rails", "Bonjour mon Amour"}

    Dim query = _
      From book In books _
      Where book.Length > 10 _
      Order By book _
      Select New With {.Book = book.ToUpper()}

    dataGridView1.DataSource = query.ToList()
  End Sub
End Class